devtools/server/actors/replay/module.js

```diff
@@ -40,6 +40,28 @@ function getWindow() {
   return gWindow;
 }
 
+function getWindowAsImageData(win) {
+  const canvas = win.document.createElementNS(
+    "http://www.w3.org/1999/xhtml",
+    "canvas"
+  );
+  const scale = getCurrentZoom(win);
+  const width = win.innerWidth;
+  const height = win.innerHeight;
+  canvas.width = width * scale;
+  canvas.height = height * scale;
+  canvas.mozOpaque = true;
+
+  const ctx = canvas.getContext("2d");
+
+  ctx.scale(scale, scale);
+  ctx.drawWindow(win, win.scrollX, win.scrollY, width, height, "#fff");
+
+  ctx.getImageData(0, 0, canvas.width, canvas.height);
+  const dataURL = canvas.toDataURL();
```

Comment:
I think it would be better if we used this call to create a JPEG data URL instead of a PNG image.  Web page PNGs can be several megabytes, whereas a JPEG with a 50% quality (which is what we use for the other screenshot data the client downloads) is normally 10x smaller.